src/logic/function/basic.lean

```diff
@@ -610,6 +610,33 @@ def set.piecewise {α : Type u} {β : α → Sort v} (s : set α) (f g : Πi, β
   Πi, β i :=
 λi, if i ∈ s then f i else g i
 
+/-! ### Bijectivity of `eq.rec`, `eq.mp`, `eq.mpr`, and `cast` -/
+
+lemma eq_rec_on_bijective {α : Sort*} {C : α → Sort*} :
+  ∀ {a a' : α} (h : a = a'), function.bijective (@eq.rec_on _ _ C _ h)
+| _ _ rfl := ⟨λ x y, id, λ x, ⟨x, rfl⟩⟩
+
+lemma eq_mp_bijective {α β : Sort*} (h : α = β) : function.bijective (eq.mp h) :=
+eq_rec_on_bijective h
+
+lemma eq_mpr_bijective {α β : Sort*} (h : α = β) : function.bijective (eq.mpr h) :=
+eq_rec_on_bijective h.symm
+
```

Comment:
Should I just remove these, now that `mp` and `mpr` are rewritten into `cast` by `simp`?

Reply:
I wouldn't remove them. One can't be sure that `simp` will be able to reach every `eq.mp`.